src/s2japp/ApplicationSettings.cc

```diff
@@ -302,6 +302,15 @@ ApplicationSettings::ApplicationSettings(int argc, const char* argv[])
                 cmd,
                 false);
 
+        TCLAP::ValueArg<std::string> argProgramWorkingDir(
+                "c",
+                "chdir",
+                "Where to chdir to before running the program",
+                false,
+                ".",
```

Comment:
I think there are some edge cases where `chdir(".")` is not the same as not chdir-ing at all.
One example I could come up with is if you don't have +x on your current directory you'll get EACCESS, though that's fine since you'd likely get EACCESS anyway when trying to execve later...

Maybe we could set the default to `""` (empty string) and then in Executor detect it and only call chdir if the argument is non-empty?

Reply:
I changed the default to `""`.